app/screens/send-bitcoin-screen/send-bitcoin-success-screen.tsx

```diff
@@ -10,38 +10,145 @@ import { useI18nContext } from "@app/i18n/i18n-react"
+import crashlytics from "@react-native-firebase/crashlytics"
+import { useHomeAuthedQuery } from "@app/graphql/generated"
+import { useIsAuthed } from "@app/graphql/is-authed-context"
 
 const SendBitcoinSuccessScreen = () => {
   const styles = useStyles()
-
+  const {
+    theme: { colors },
+  } = useTheme()
+  const [isActive, setIsActive] = React.useState(false)
+  const isAuthed = useIsAuthed()
+  const [showImprovement, setshowImprovement] = React.useState(false)
+  const [improvement, setImprovement] = React.useState("")
   const navigation =
     useNavigation<StackNavigationProp<RootStackParamList, "sendBitcoinSuccess">>()
 
+  const { data: dataAuthed } = useHomeAuthedQuery({
+    skip: !isAuthed,
+    fetchPolicy: "network-only",
+    errorPolicy: "all",
+    // this enables offline mode use-case
+    nextFetchPolicy: "cache-and-network",
+  })
+  const submitImprovement = async () => {
+    navigation.popToTop()
+    setshowImprovement(false)
+    // await fetch("https://chat.galoy.io/api/v4/posts", {
+    //   method: "POST",
+    //   headers: {
+    //     "Content-Type": "application/json",
+    //     "Authorization": `bearer personal_access_token`,
+    //   },
+    //   body: JSON.stringify({
+    //     // eslint-disable-next-line camelcase
+    //     channel_id: "n59hg9abetdrtygof11kncjbdw",
+    //     message: improvement,
+    //   }),
+    // }).then(() => {
+    //   setshowImprovement(false)
+    //   navigation.popToTop()
+    // })
+  }
+
+  const dismiss = () => {
+    setIsActive(false)
+    setshowImprovement(true)
+  }
+
+  const dismissSuggestionModal = () => {
+    navigation.popToTop()
+    setshowImprovement(false)
+  }
+
+  const rateUs = () => {
+    Rate.rate(ratingOptions, (success, errorMessage) => {
+      if (success) {
+        crashlytics().log("User went to the review page")
+      }
+      if (errorMessage) {
+        crashlytics().recordError(new Error(errorMessage))
+      }
+    })
+  }
+
   const { LL } = useI18nContext()
+  const FEEDBACK_DELAY = 3000
+  const transactionsEdges =
+    dataAuthed?.me?.defaultAccount?.transactions?.edges ?? undefined
   const CALLBACK_DELAY = 3000
   useEffect(() => {
+    if (transactionsEdges?.length) {
+      const feedbackTimeout = setTimeout(() => setIsActive(true), FEEDBACK_DELAY)
+      return () => {
+        clearTimeout(feedbackTimeout)
+      }
+    }
+
     const navigateToHomeTimeout = setTimeout(navigation.popToTop, CALLBACK_DELAY)
     return () => clearTimeout(navigateToHomeTimeout)
-  }, [navigation])
+  }, [transactionsEdges, navigation])
+
+  useEffect(() => {
+    if (isActive) {
+      Alert.alert(LL.support.enjoyingApp(), "", [
+        {
+          text: LL.common.No(),
+          onPress: () => dismiss,
+        },
+        {
+          text: LL.common.yes(),
+          onPress: () => rateUs,
+        },
+      ])
+    }
+  }, [isActive, LL])
+
+  const showSuggestionModal = () => {
+    return (
+      <Modal
```

Comment:
Can you make this it's own component?
Having so much code in the screen for suggestion seems counterintuitive to me - it's better to have this as it's own component.

Perhaps make the component auto display itself if the user is receiving for the first time as well?

And the local auth query must also reside there.

The send screen as well as the home screen can just have `<Suggestion ... />` in that way.